float64[] semi_axes
float64[] center
#float64 epsilon
float64 angle